Cloudwatch: Panic on query

Not sure what I did. Been clicking around the cheat sheet examples.
grafana-server ended with a panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4ef5189]

goroutine 1189 [running]:
github.com/grafana/grafana/pkg/tsdb/cloudwatch.generateGroupKey(0xc001177940, 0x2, 0x2, 0x15, 0xc0011d0188, 0x0)
	/Users/david/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/log_query.go:116 +0x29
github.com/grafana/grafana/pkg/tsdb/cloudwatch.groupResults(0xc0010da980, 0xc000da6180, 0x2, 0x2, 0x2, 0x2, 0x0, 0x0, 0xc0009ec2d0)
	/Users/david/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/log_query.go:95 +0x3f1
github.com/grafana/grafana/pkg/tsdb/cloudwatch.(*CloudWatchExecutor).executeLogActions.func1(0x1, 0x0)
	/Users/david/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/log_actions.go:43 +0x2d7
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc000c5c780, 0xc000e37040)
	/Users/david/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
	/Users/david/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:54 +0x66
```

--- a/pkg/tsdb/cloudwatch/log_query.go
+++ b/pkg/tsdb/cloudwatch/log_query.go
@@ -8,14 +8,13 @@ import (
 )
 
 func logsResultsToDataframes(response *cloudwatchlogs.GetQueryResultsOutput) (*data.Frame, error) {
-	rowCount := len(response.Results)
 	fieldValues := make(map[string]interface{})
 
 	// Maintaining a list of field names in the order returned from CloudWatch
 	// as just iterating over fieldValues would not give a consistent order
 	fieldNames := make([]*string, 0)
 
-	for i, row := range response.Results {
+	for _, row := range response.Results {
 		for _, resultField := range row {
 			// Strip @ptr field from results as it's not needed
 			if *resultField.Field == "@ptr" {
@@ -27,9 +26,9 @@ func logsResultsToDataframes(response *cloudwatchlogs.GetQueryResultsOutput) (*d
 
 				// Check if field is time field
 				if _, err := time.Parse(cloudWatchTSFormat, *resultField.Value); err == nil {
-					fieldValues[*resultField.Field] = make([]*time.Time, rowCount)
+					fieldValues[*resultField.Field] = make([]*time.Time, 0)
 				} else {
-					fieldValues[*resultField.Field] = make([]*string, rowCount)
+					fieldValues[*resultField.Field] = make([]*string, 0)
 				}
 			}
 
@@ -39,9 +38,9 @@ func logsResultsToDataframes(response *cloudwatchlogs.GetQueryResultsOutput) (*d
 					return nil, err
 				}
 
-				timeField[i] = &parsedTime
+				fieldValues[*resultField.Field] = append(timeField, &parsedTime)
 			} else {
-				fieldValues[*resultField.Field].([]*string)[i] = resultField.Value
+				fieldValues[*resultField.Field] = append(fieldValues[*resultField.Field].([]*string), resultField.Value)
 			}
 		}
 	}

--- a/pkg/tsdb/cloudwatch/log_query_test.go
+++ b/pkg/tsdb/cloudwatch/log_query_test.go
@@ -78,6 +78,8 @@ func TestLogsResultsToDataframes(t *testing.T) {
 					Value: aws.String("fakelog"),
 				},
 			},
+			// Sometimes cloudwatch returns empty row
+			{},
 			{
 				&cloudwatchlogs.ResultField{
 					Field: aws.String("@ptr"),

